MVP Module/Source/Feed Presentation/FeedErrorViewModel.swift

```
@@ -0,0 +1,11 @@
+//
+//  FeedErrorViewModel.swift
+//  MVPTests
+//
+//  Created by Tak Mazarura on 18/03/2021.
+//  Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+struct FeedErrorViewModel {
+	let hasError: Bool
```

Choosing which message to show is a presentation decision - it should be done by the presenter. 

So the `message ` could be an optional `String?` instead of a boolean. This way, Presenter can decide which message to show and when to show/hide the message.

For example, in the `didStartLoadingFeed()` method, the Presenter can call  `feedErrorView.display(FeedErrorViewModel(message: .none))` to hide the error.

And in `didFinishLoadingFeed(with error:)`, the Presenter can call  `feedErrorView.display(FeedErrorViewModel(message: Localized.Feed.loadError))` to show the error.

This way, the ViewController doesn't need to decide which error to show and when to hide the error. It's often easier to manage state when all changes are centralized.

---

Ahh, ok, so the `ViewModel` in the MVP is all data and no behaviour, in addition the data should be in a state where it is as ready for presentation as possible such that the View does not need to have any logic for presentation as done in my previous implementation.

---

Updated! :)